crates/wastebin_core/src/expiration.rs

```diff
@@ -62,73 +94,81 @@ impl Display for Expiration {
             (r > 0).then_some((r, dividend % divisor))
         }
 
+        const SEPARATOR: &str = ", ";
+        let mut sep = "";
         let mut secs = self.duration.as_secs();
 
         if secs == 0 {
             return write!(f, "never");
         }
 
-        let mut parts = Vec::new();
-
-        if let Some((years, rem)) = div(secs, 60 * 60 * 24 * 7 * 4 * 12) {
+        if let Some((years, rem)) = div(secs, YEAR_SECS) {
             if years > 1 {
-                parts.push(format!("{years} years"));
+                write!(f, "{years} years")?;
             } else {
-                parts.push(String::from("1 year"));
+                write!(f, "1 year")?;
```

Comment:
Could you please drop these changes. The `join` in the original code already inserts separators.